readthedocs/projects/exceptions.py

```diff
@@ -60,3 +60,10 @@ class ProjectSpamError(Exception):
     This error is not raised to users, we use this for banning users in the
     background.
     """
+
+
+class InvalidParamsException(Exception):
+
+    """Error raised when incorrect parameters are passed to a function/class."""
```

Comment:
We can use ValueError and TypeError in most of the cases of this custom Exception I think, feel free to create another PR with this changes

 https://docs.python.org/3/library/exceptions.html#ValueError
https://docs.python.org/3/library/exceptions.html#TypeError

Reply:
@stsewd 
Thanks for this.
I'll be opening another PR about this soon.